be2-scala/src/main/scala/ch/epfl/pop/json/MessageDataProtocol.scala

```diff
@@ -37,8 +41,31 @@ object MessageDataProtocol extends DefaultJsonProtocol {
   }
 
 
+  // ------------------------------- METADATA UTILITY -------------------------------------- //
+  // retrieve information from the header if it is correct.
+  //
+  // Succeeds if both 'object' and 'action' are present and are strings
+  def parseHeader(data: String): Try[(ObjectType, ActionType)] =
+    Try {
+      data.parseJson.asJsObject.getFields(PARAM_OBJECT, PARAM_ACTION) match {
+        case Seq(objectString @ JsString(_), actionString @ JsString(_)) =>
+          (objectString.convertTo[ObjectType], actionString.convertTo[ActionType])
```

Comment:
I don't exactly know whether this is really a relevant point, as it's wrapped in the Try, but I'd still add a case _ => just to be sure

Reply:
That is absolutely right, fixed it up!